sdks/python/apache_beam/runners/portability/stager_test.py

```diff
@@ -420,134 +375,167 @@ def test_sdk_location_local_directory_not_present(self):
+          self.__module__, TestStager.__name__,
+          TestStager._download_file.__name__
+      ])):
+        self.assertEqual([names.DATAFLOW_SDK_TARBALL_FILE],
+                         self.stager.stage_job_resources(
+                             options, staging_location=staging_dir))
+
+  def test_sdk_location_remote_wheel_file(self):
     staging_dir = self.make_temp_dir()
     sdk_filename = 'apache_beam-1.0.0-cp27-cp27mu-manylinux1_x86_64.whl'
-    sdk_location = 'gs://my-gcs-bucket/' + sdk_filename
+    sdk_location = '/tmp/remote/my-bucket/' + sdk_filename
 
     options = PipelineOptions()
-    options.view_as(GoogleCloudOptions).staging_location = staging_dir
     self.update_options(options)
     options.view_as(SetupOptions).sdk_location = sdk_location
 
-    with mock.patch('apache_beam.runners.dataflow.internal.'
-                    'dependency._dependency_file_copy'):
-      self.assertEqual(
-          [sdk_filename],
-          dependency.stage_job_resources(options))
+    # We can not rely on actual remote file systems paths hence making
+    # '/tmp/remote/' a new remote path.
+    def is_remote_path(dummy_self, path):
+      return path.startswith('/tmp/remote/')
+
+    with mock.patch('.'.join([
+        self.__module__, TestStager.__name__, TestStager.stage_artifact.__name__
+    ])):
+      with mock.patch('.'.join([
+          self.__module__, TestStager.__name__,
+          TestStager._download_file.__name__
+      ])):
+        with mock.patch(
+            '.'.join([
+                self.__module__, TestStager.__name__,
+                TestStager._is_remote_path.__name__
+            ]), is_remote_path):
+          self.assertEqual([sdk_filename],
+                           self.stager.stage_job_resources(
+                               options, staging_location=staging_dir))
 
   def test_sdk_location_http(self):
     staging_dir = self.make_temp_dir()
     sdk_location = 'http://storage.googleapis.com/my-gcs-bucket/tarball.tar.gz'
 
     options = PipelineOptions()
-    options.view_as(GoogleCloudOptions).staging_location = staging_dir
     self.update_options(options)
     options.view_as(SetupOptions).sdk_location = sdk_location
 
-    def file_download(_, to_folder):
-      tarball_path = os.path.join(to_folder, 'sdk-tarball')
-      with open(tarball_path, 'w') as f:
+    def file_download(dummy_self, _, to_path):
+      with open(to_path, 'w') as f:
         f.write('Package content.')
-      return tarball_path
+      return to_path
 
-    with mock.patch('apache_beam.runners.dataflow.internal.'
-                    'dependency._dependency_file_download', file_download):
-      self.assertEqual(
-          [names.DATAFLOW_SDK_TARBALL_FILE],
-          dependency.stage_job_resources(options))
-
-    tarball_path = os.path.join(
-        staging_dir, names.DATAFLOW_SDK_TARBALL_FILE)
+    with mock.patch(
+        '.'.join([
+            self.__module__, TestStager.__name__,
```

Comment:
Please use human-readable path to the method in the first argument of mock.patch() throughout this file.

Reply:
The string makes it harder to refactor code and refactor things. I would rather prefer modules over string.